File: ListadoPersonasYDepartamentos/Domain/Entities/Persona.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Persona
    {
        #region atributos privados
        private int _id;
        private string _nombre;
        private string _apellido;
        private DateTime _fechaNac;

        private string _direccion;

        private string _telefono;
        private string _imagen;
        private int _idDepartamento;

        #endregion
        #region getters y setters
        public int id 
        {
            get { return _id; } 
            set { _id = value; }
        }
        public string nombre
        {
            get { return _nombre; }
            set { _nombre = value; }
        }
        public string apellido
        {
            get { return _apellido; }
            set { _apellido = value; }
        }

        public DateTime fechaNac
        {
            get { return _fechaNac; }
            set { _fechaNac = value; }
        }

        public string direccion
        {
            get { return _direccion; }
            set { _direccion = value; }
        }

        public string telefono
        {
            get { return _telefono; }
            set { _telefono = value; }
        }

        public string imagen
        {
            get { return _imagen; }
            set { _imagen = value; }
        }
        public int idDepartamento
        {
            get { return _idDepartamento; }
            set { _idDepartamento = value; }
        }
        #endregion
        #region constructores
        public Persona() { }
        public Persona(int id, string nombre, string apellido)
        {
            _id = id;
            _nombre = nombre;
            _apellido = apellido;
        }
        public Persona(int id, string nombre, string apellido,
               string direccion, string telefono, string imagen, int idDepartamento)
        {
            _id = id;
            _nombre = nombre;
            _apellido = apellido;
            _direccion = direccion;
            _telefono = telefono;
            _imagen = imagen;
            _idDepartamento = idDepartamento;
        }

        #endregion
    }
}

File: tema8/ListadoPersonasCRUD/src/Domain/Entities/Persona.ts
export class Persona {
  //#region Atributos privados
  private _id: number;
  private _nombre: string;
  private _apellidos: string;
  private _fechaNac: Date;
  private _direccion: string;
  private _telefono: string;
  private _foto: string;
  private _idDepartamento: number;
  //#endregion

  //#region Constructores
  constructor();
  constructor(
    id: number,
    nombre: string,
    apellidos: string,
    fechaNac: Date,
    direccion: string,
    telefono: string,
    foto: string,
    idDepartamento: number
  );
  constructor(
    id?: number,
    nombre?: string,
    apellidos?: string,
    fechaNac?: Date,
    direccion?: string,
    telefono?: string,
    foto?: string,
    idDepartamento?: number
  ) {
    this._id = id || 0;
    this._nombre = nombre || "";
    this._apellidos = apellidos || "";
    this._fechaNac = fechaNac || new Date();
    this._direccion = direccion || "";
    this._telefono = telefono || "";
    this._foto = foto || "";
    this._idDepartamento = idDepartamento || 0;
  }
  //#endregion

  //#region Getters y Setters
  public get id(): number {
    return this._id;
  }

  public set id(value: number) {
    this._id = value;
  }

  public get nombre(): string {
    return this._nombre;
  }

  public set nombre(value: string) {
    this._nombre = value;
  }

  public get apellidos(): string {
    return this._apellidos;
  }

  public set apellidos(value: string) {
    this._apellidos = value;
  }

  public get fechaNac(): Date {
    return this._fechaNac;
  }

  public set fechaNac(value: Date) {
    this._fechaNac = value;
  }

  public get direccion(): string {
    return this._direccion;
  }

  public set direccion(value: string) {
    this._direccion = value;
  }

  public get telefono(): string {
    return this._telefono;
  }

  public set telefono(value: string) {
    this._telefono = value;
  }

  public get foto(): string {
    return this._foto;
  }

  public set foto(value: string) {
    this._foto = value;
  }

  public get idDepartamento(): number {
    return this._idDepartamento;
  }

  public set idDepartamento(value: number) {
    this._idDepartamento = value;
  }
  //#endregion
}